### DISCUSSION TITLE
Cone Detection for Formula Student Driverless Competition (Project 248)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#86] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project 248](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/blob/main/projects/Cone%20Detection%20for%20Formula%20Student%20Driverless%20Competition)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by Kaushik-C** [comment]
Hi, I'm Kaushik and I'm a student at University of Kassel and interested in working and collaborating on this project.

**Reply by veeralakshendra** [reply]
Hi Kaushik,
I appreciate your interest. Please feel free to go through the resources list added to the problem statement. The most helpful one is [YOLOv2 Object Detection: Data Labelling to Neural Networks in MATLAB](https://blogs.mathworks.com/student-lounge/2020/07/07/yolov2-object-detection-data-labelling-to-neural-networks-in-matlab/). Please let me know if you have any queries. Also, you can refer to the solution that has been accepted:  [solution by Vivin](https://github.com/vivinvarshans/Cone-Detection-MathWork-Assignment)

**Comment by hemu312** [comment]
Hi, I'm Hemant Kumar Meena. I am a part time freelancer. I am interested in Linux, robotics.
I am working on this project

**Comment by hemu312** [comment]
Guys, Is this project doable on matlab linux.?
Because when I try to run skidpad test example I receive error.

**Reply by veeralakshendra** [reply]
Hi Hemu,

Thank you for the query. Yes, as the model will open in Unreal, the skidpad model will through the error in Linux. However, the model is just used to collect the camera view data in video format that can be used to develop the detection algorithm. We can provide you with video files from different camera locations. Do you want us to share the video files?

**Reply by hemu312** [reply]
yes sir that will be great

**Reply by veeralakshendra** [reply]
Hi Hemu,
Sorry for the delay. Please find the video file attached. 

https://github.com/mathworks/MATLAB-Simulink-Challenge-Project-Hub/assets/62535885/8b0f5624-1774-4dfa-8d83-e27db1a41622

**Reply by hemu312** [reply]
thanks sir

**Comment by veeralakshendra** [comment]
We are glad to share that we have received our first solution for this project. Congratulations to Vivin Varshan for completing the project. Please find his solution here: [solution by Vivin](https://github.com/vivinvarshans/Cone-Detection-MathWork-Assignment). You can use this as a reference.

**Comment by hemu312** [comment]
Sir, I got to know that yolov2 works best with input image size of 416x416.
So, I am doing crop(to 720x720) and then resize(to 416x416).
Am I allowed to do that ?

**Reply by veeralakshendra** [reply]
That's correct, Hemu. 416x416 is preferred for YOLO. You can do that programmatically. Check out this video which demonstrates this process (timestamp: 1:55 to 3:45): [video data processing](https://youtu.be/g_Vj1ASBcYo?si=kF6URHUGKsmhvb7V&t=115)

**Comment by DkPanseriya** [comment]
I developed and trained my network on Mac and later had to use Windows for simulating with Unreal engine. I was so focused on the training part, that I didn't realize the Skidpad model in Vehicle Dynamics Blockset has Orange cones. I trained my network for the real event setup of Blue and Yellow cones. 
I will submit my assignment for the detection of blue and yellow cones and will try to attach a testing video (on some other track video from dashcams) for the evaluation.
Hope that would be understandable and acceptable.

**Reply by hemu312** [reply]
I think that problem can be solved if we convert images first to grayscale.

**Reply by veeralakshendra** [reply]
Hi Dk,
Thank you for your query. No problem. You can surely submit the project with yellow and blue cones. Please make sure you add the instructions to run the code in the readme. Looking forward to your solution.

**Comment by atindra2504** [comment]
"Hi, I am a student at Ramaiah Institute of Technology and interested in working and collaborating on this project
